### PR TITLE
chore: upgrade node version to >=20 and other minor improvements

### PR DESCRIPTION
**Pull request by kilted-andres**
## fixes part of KILTProtocol/ticket#3081

- Makes sure the `package.json` specifies `pnpm` and `node` version. 
- Upgrades the required **node version**  on the `package.json` to **>=20**.
- Upgrades the used **node version**  on the `.nvmrc`  to `lts/Iron`. 
- Makes sure the **GitHub workflows** uses node version 20. 
  For this following must be set: 
  - `actions/checkout@v4`
  - `actions/setup-node@v4` **with:**  _either_
    - **node-version-file:** '`package.json`' or '`.nvmrc`' 
    -  **node-version:**  `'`<Node.js version using [SemVer](https://semver.org/) or [aliases](https://github.com/nodejs/Release?tab=readme-ov-file#release-schedule) > `'`
  -  `aws-actions/configure-aws-credentials@v4`
  -  `aws-actions/amazon-ecr-login@v2`
  -  `aws-actions/configure-aws-credentials@v4`
 - Makes sure the **Dockerfiles** also use the node version 20. 
  This means that the following must be set: 
      - `FROM node:20-alpine as builder`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:20.14.0-alpine AS base
+FROM node:20-alpine AS base
```
Comment (arty-name) [inline Dockerfile:1]: Would renovate overwrite this with a specific version later on?

File: package.json [hunk 2]
```diff
@@ -2,8 +2,9 @@
   "repository": "git@github.com:galaniprojects/ctypehub.git",
   "license": "BSD-4-Clause",
   "type": "module",
+  "packageManager": "pnpm@9.6.0+sha512.38dc6fba8dba35b39340b9700112c2fe1e12f10b17134715a4aa98ccf7bb035e76fd981cf0bb384dfa98f8d6af5481c2bef2f4266a24bfa20c34eb7147ce0b5e",
```
Comment (arty-name) [inline package.json:5]: I assume this line was added by corepack automatically. We can make it shorter:
```suggestion
  "packageManager": "pnpm@9.6.0",
```

**Review by arty-name — APPROVED** [review]
Since we’re touching pnpm in this PR, it could be a good moment to cleanup some things about it. I believe Docker and CI are not using corepack at the moment.

File: Dockerfile (hunk 1 above)
Comment (kilted-andres) [reply Dockerfile:1]: If the bot overwrites it afterwards, it is ok.

Comment (arty-name) [reply Dockerfile:1]: I hope they’ve fixed the issue where this tag would mean "whatever matching version was cached on this host in the last months"

Comment (kilted-andres) [reply Dockerfile:1]: Not sure about that either. 😅

**Pull request merged**